src/cargo/util/paths.rs

```diff
@@ -89,6 +89,20 @@ pub fn normalize_path(path: &Path) -> PathBuf {
     ret
 }
 
+#[cfg(unix)]
```

Comment:
If we get rid of the `is_executable` check I can move these back. Also I have no idea if this was the right place to put them but I had no idea how to call functions from a bin in a lib in the same crate, I'm assuming its not something you can do.